Tasks/AzureFileCopy/AzureUtilityGTE1.0.ps1

```diff
@@ -340,4 +340,108 @@ function Remove-AzureMachineCustomScriptExtension
     }
 
     return $response
+}
+
+function Add-NetworkSecurityRuleConfig
+{
+    param([object]$securityGroup,
+          [string]$ruleName,
+          [string]$rulePriotity,
+          [string]$winrmHttpsPort)
+
+    if($securityGroup)
+    {
+        $securityGroupName = $securityGroup.Name
+        try
+        {
+            Write-Verbose -Verbose "[Azure Call]Getting network security rule config $ruleName under security group $securityGroupName"
+            $winRMConfigRule = Get-AzureRmNetworkSecurityRuleConfig -NetworkSecurityGroup $securityGroup -Name $ruleName -EA SilentlyContinue
+            Write-Verbose -Verbose "[Azure Call]Got network security rule config $ruleName under security group $securityGroupName"
+        }
+        catch
+        {
+            #Ignore the exception
+        }
+
+        # Add the network security rule if it doesn't exists
+        if(-not $winRMConfigRule)
+        {            
+            for($i=1; $i -le 3; $i++)
+            {
+                try
+                {
+                    Write-Verbose -Verbose "[Azure Call]Adding inbound network security rule config $ruleName with priority $rulePriotity for port $winrmHttpsPort under security group $securityGroupName"
+                    $result = Add-AzureRmNetworkSecurityRuleConfig -NetworkSecurityGroup $securityGroup -Name $ruleName -Direction Inbound -Access Allow -SourceAddressPrefix '*' -SourcePortRange '*' -DestinationAddressPrefix '*' -DestinationPortRange $winrmHttpsPort -Protocol * -Priority $rulePriotity | Set-AzureRmNetworkSecurityGroup
+                    Write-Verbose -Verbose "[Azure Call]Added inbound network security rule config $ruleName with priority $rulePriotity for port $winrmHttpsPort under security group $securityGroupName"                         
+                }
+                catch
+                {           
+                    $newPort = [convert]::ToInt32($rulePriotity, 10) + 50;
```

Comment:
Should this responsibility be with caller? In other words should this method take a list of ports and attempt to apply them. You can then have flexibility in callers to pass any list of port that they think is right for adding. In this case, Azure file copy can pass 3 choices with + 50 difference. Thoughts?